packages/delegation-process/src/services/validators.ts

```diff
@@ -131,10 +128,26 @@ export const assertIsState = (
   delegation: Delegation
 ): void => {
   if (delegation.state !== state) {
-    throw incorrectState(
-      delegation.id,
-      delegation.state,
-      delegationState.waitingForApproval
-    );
+    throw incorrectState(delegation.id, delegation.state, state);
+  }
+};
+
+export const assertIsOneOfStates = (
```

Comment:
Instead of creating another function, how about refactor `assertIsState` with the support of a single state or an array of states? So that we can rely on a single function to check the states